### PR TITLE
RDAPI-22426 YAMLES has a more strict e18n syntax

### PR DESCRIPTION
**Pull request by bordigoni**
Thank you for your contribution to the Axway-Open-Docs repo.

https://deploy-preview-1508--axway-open-docs.netlify.app/docs/apim_yamles/apim_yamles_references/yamles_known_limitations/

## Describe the changes

* Syntax is more strict now for `{{ }}` placeholders
* This allows remove know limitations
* Some other fixes added related to this change
* Some other minor fixes

## Checklist for contributors

Before submitting this PR, please make sure:

* [ ] You have read the [contribution guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/)
* [ ] You have signed the [Axway CLA](https://cla.axway.com/)
* [ ] You have verified the technical accuracy of your change
* [ ] You have verified that your change does not expose any sensitive information (passwords, keys, etc.)
* [ ] You have followed the [Markdown guidelines](https://axway-open-docs.netlify.com/docs/contribution_guidelines/writing_markdown/)  (unless this is is a Netlify CMS contribution)
* [ ] You have verified that all status checks have passed

_Put an x in the boxes that apply. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your change._

### CODE REVIEW

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 1]
```diff
@@ -125,3 +125,30 @@ You can run the following help command for more details on each parameter:
 ```
 yamles change-passphrase --help
 ```
+
+## Change the encryption passphrase of an environmentalized YAML configuration
+
+All the commands above remains valid.
+
+When encrypted data reside in values.yaml it will also be changed, unless it is an environment variable.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:133]: ....resides in `values.yaml`.....

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 2]
```diff
@@ -125,3 +125,30 @@ You can run the following help command for more details on each parameter:
 ```
 yamles change-passphrase --help
 ```
+
+## Change the encryption passphrase of an environmentalized YAML configuration
+
+All the commands above remains valid.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:131]: remain

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 3]
```diff
@@ -125,3 +125,30 @@ You can run the following help command for more details on each parameter:
...
+
+When encrypted data reside in values.yaml it will also be changed, unless it is an environment variable.
+
+```yaml
+# Entity file
+...
+fields:
+  # stays pristine
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:139]: # remain unchanged

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 4]
```diff
@@ -125,3 +125,30 @@ You can run the following help command for more details on each parameter:
...
+  someCipheredKey: eW91c2hhbGxub3RwYXNzeW91Zm9vbGlzaHBpcmF0ZQ==
+...
+
+# values.yaml
+---
+# will be changed
+conf:  
+  password: wWVn7dS/ycwDg7Miqd1TU0YKTiOY//5i
+# stays pristine
+ldap:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:152]: remain unchanged

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md (hunk 1 above)
Comment (andreamussap) [reply content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:133]: @bordigoni please leave Melissas fback with me, I'm already reviewing this PR

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md [hunk 5]
```diff
@@ -23,14 +23,14 @@ We assume:
...
-type: "DbConnection"
+type: DbConnection
 fields:
    name: MySQL
-   maxIdle: {{ db.maxIdle }}
-   maxWait: {{ foo }}
-   url: "jdbc:mysql://{{ db.host }}:3306/DefaultDb"
-   password: {{ db.password }}
-   username: "{{ db.username }}"
+   maxIdle: '{{ db.maxIdle }}'
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:29]: I like these spaces after {{ and before }}
But if handlebars is rewriting them out, our examples should not have them

Comment (bordigoni) [reply content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:29]: Agreed. Still need to check if it can be changed in handlesbar, I suspect not.

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 6]
```diff
@@ -6,7 +6,7 @@
 "description": "Learn how to use the YAML configuration CLI to encrypt, decrypt, and re-encrypt a YAML configuration."
 }
 
-These are the encryption related options available in the YAML CLI:
+The YAML CLI provides two encryption options:
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:9]: @bordigoni @mcollins-axway I've just pushed one commit to this file and I've made small changes to improve readability. Please review if I didn't get anything wrong

https://deploy-preview-1508--axway-open-docs.netlify.app/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption/

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 7]
```diff
@@ -74,47 +74,51 @@ fields:
...
+A private key is sensitive data that is externalized into a separate file in the YAML configuration. Its data is also of type `encrypted`, and the file content must be encrypted if the YAML configuration is encrypted. A private key file can be encrypted as follows:
 
 ```
 ./yamles encrypt --file /home/user/private-key.der --passphrase "changeme"
 ```
 
 See section [Add a new certificate and private key to a YAML configuration](/docs/apim_yamles/yamles_edit/#add-a-new-certificate-and-private-key-to-a-yaml-configuration) for more information regarding adding a certificate and private key to a YAML configuration. Note that `encrypt --file` option may be used to encrypt the content of any type of file. The operation will update the file which will always have binary content after encryption. Note that encrypting a private key via `openssl` and adding it to the YAML configuration is not supported.
 
-## Change the encryption passphrase of a YAML configuration
+## Change the encryption passphrase
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:85]: @bordigoni I've made changes to this section to improve clarity. There was no need for 2 subsections, and the heading was unnecessary long. However, the last part ("Encrypted data in the values.yaml file....") isn't clear to me. I think it deserves some rewording.

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md (hunk 5 above)
Comment (bordigoni) [reply content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:29]: No way I can change that. I've updated the docs

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md (hunk 7 above)
Comment (bordigoni) [reply content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:85]: reworded

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 8]
```diff
@@ -74,47 +74,51 @@ fields:
...
 
-If you have encrypted your YAML configuration you may wish to encrypt it with a different passphrase. You may also need to decrypt it by setting the `--new-passphrase` to `""`. In order to do this you need to know what the current passphrase is.
-
-It is also possible to change the passphrase of a YAML configuration at deployment time via [projdeploy](/docs/apim_reference/devopstools_ref/#projdeploy-command-options).
+To change the passphrase of an encrypted YAML configuration or set the passphrase to null, you must know your current passphrase.
 
 The following are examples of how you can use the `change-passphrase` option in the `yamles` CLI to change the passphrase of a YAML configuration:
 
-**Example 1**: Specify a directory from `changeme` to `newpassphrase`:
+**Example**: Change the passphrase from `changeme` to null:
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:91]: @bordigoni did you see that I've added this example? Is it correct?

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 9]
```diff
@@ -15,28 +15,34 @@ These are the encryption related options available in the YAML CLI:
 
 Encryption of XML or YAML configuration means that sensitive fields of type `encrypted` in the Entity Store model are encrypted. All other data remains in the clear. If the XML federated configuration you converted to YAML was not encrypted, it remains unencrypted after conversion. If you wish to deploy the configuration to an API Gateway group that has a group passphrase set, you must encrypt the YAML configuration before deployment.
 
-If you get an error when attempting to encrypt the YAML configuration, that indicates that the passphrase for the configuration is incorrect, it means that the configuration is already encrypted. Use the `change-passphrase` option to change the passphrase.
+If you get an error when attempting to encrypt the YAML configuration, this means that the passphrase for the configuration is incorrect or that the configuration is already encrypted. Use the `change-passphrase` option to change the passphrase.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:18]: The meaning of this sentence has changed in a subtle manner I think and I think is less accurate. Perhaps the original is confusing. I will try again...

The `encrypt` option is intended to encrypt a YAML configuration that is not yet encrypted. If an attempt to `encrypt` results in a error indicating an incorrect passphrase, this means that the YAML configuration is already encrypted. Use the `change-passphrase` option to change the passphrase.

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 10]
```diff
@@ -47,7 +53,7 @@ yamles encrypt --help
...
+If the XML federated configuration you converted to YAML was encrypted with a passphrase, it remains encrypted with the same passphrase after conversion to YAML format. If you wish to add new configuration that includes fields of type `encrypted`, you need to encrypt that data with the same entity store passphrase so that it can be read by the API Gateway when deployed. For example, you wish to add an entity of type `DbConnection`, which has a `password` field, of `encrypted` type. You must put the encrypted value of the password string into the YAML file for the `DbConnection`. If your database password is `dbpassword` and your passphrase for the YAML configuration is `changeme`.
 
 Use the following command to encrypt the database password string value:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:58]: This looks wrong, need to merge sentence as follows...

"If your database password is `dbpassword` and your passphrase for the YAML configuration is `changeme`, use the following command to encrypt the database password string value:"

What we have there at the moment does not make sense unless these sentences are together.

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 11]
```diff
@@ -74,54 +80,93 @@ fields:
...
 
-If you have encrypted your YAML configuration you may wish to encrypt it with a different passphrase. You may also need to decrypt it by setting the `--new-passphrase` to `""`. In order to do this you need to know what the current passphrase is.
-
-It is also possible to change the passphrase of a YAML configuration at deployment time via [projdeploy](/docs/apim_reference/devopstools_ref/#projdeploy-command-options).
+To change the passphrase of an encrypted YAML configuration you must know your current passphrase.
 
 The following are examples of how you can use the `change-passphrase` option in the `yamles` CLI to change the passphrase of a YAML configuration:
 
-**Example 1**: Specify a directory from `changeme` to `newpassphrase`:
+**Example**: Specify a directory to change from `changeme` to `newpassphrase`:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:97]: .... to change the passphrase from ....

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 12]
```diff
@@ -74,54 +80,93 @@ fields:
...
 
 ```
 ./yamles change-passphrase --source /home/user/yaml --old-passphrase changeme --new-passphrase newpassphrase
 ```
 
-**Example 2**: Specify a URL from `changeme` to `newpassphrase`:
+**Example**: Specify a URL to change from `changeme` to `newpassphrase`:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:103]: .... to change the passphrase from ....

File: content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md [hunk 13]
```diff
@@ -74,54 +80,93 @@ fields:
...
 
 ```
-./yamles change-passphrase --source yaml:file:/home/user/apiprojects/myyaml --old-passphrase changeme --new-passphrase newpassphrase
+./yamles change-passphrase --source yaml:file:/home/user/yaml --old-passphrase changeme --new-passphrase newpassphrase
 ```
 
-**Example 3**: Specify `.tar.gz` file from `changeme` to `newpassphrase`:
+**Example**: Specify `.tar.gz` file to change from `changeme` to `newpassphrase`:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_cli/yamles_cli_encryption.md:109]: .... to change the passphrase from ....

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md [hunk 14]
```diff
@@ -40,21 +37,21 @@ fields:
...
 # Processed result (in-memory at load time)
 ---
-type: "DbConnection"
+type: DbConnection
 fields:
    name: MySQL
    maxIdle: 1
-   maxIdle: {{ foo }}                      # <--- error
-   url: "jdbc:mysql://staging.db.acme.com:3306/DefaultDb"
+   maxIdle: '{{foo}}'                    # <--- error
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:51]: This should be maxWait as it is foo above

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md [hunk 15]
```diff
@@ -8,29 +8,26 @@
...
+* System environment variable with and without default value.
+* Encrypted passwords.
+* Conservation of other environmentalization methods.
+* An unresolved placeholder.
 
-We assume:
-
-* `ES_DB_PASSWORD` is set with a value encrypted by a passphrase
-* `ES_DB_HOST` is not set
+In the following example snippet, we assume that `ES_DB_PASSWORD` is set with a value encrypted by a passphrase, and `ES_DB_HOST` is not set.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:18]: Can we make clear that ES_* are system environment variables.

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md [hunk 16]
```diff
@@ -40,21 +37,21 @@ fields:
...
 ```yaml
 # Processed result (in-memory at load time)
 ---
-type: "DbConnection"
+type: DbConnection
 fields:
    name: MySQL
    maxIdle: 1
-   maxIdle: {{ foo }}                      # <--- error
-   url: "jdbc:mysql://staging.db.acme.com:3306/DefaultDb"
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_environmentalization_example.md:55]: Add comment to say default is used as env variable not set ?

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 17]
```diff
@@ -6,42 +6,48 @@
 "description": "Learn how to fix errors that might occur when validating your YAML configuration."
 }
 
-This section covers examples of issues that may be found using the `validate` option of the `yamles` CLI tool.
-All ERRORs are shown on stdout.
+This section covers examples of issues that might occurs while using the `validate` option of the `yamles` CLI tool. All errors are shown on `stdout`.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:9]: that might "occur" not "occurs"

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 18]
```diff
@@ -6,42 +6,48 @@
...
+```
+entity=/External Connections/DB Connections/MySQL/local, type=DbConnection, field='maxWait', file=/home/user/yaml/External Connections/DB Connections/MySQL.yaml, message='Expected fo find data of type <long> but value is <9999999999999999999999999999999999>'
+```
 
-Look in the file named `/home/user/yaml/External Connections/DB Connections/MySQL.yaml`. It contains an entity of type `DbConnection` named `local` with a field named `maxWait`. Its value is set to `9999999999999999999999999999999999` but a long is expected by the model. Fix the field value to be an valid long. Max long value is `9.223372e+18 2⁶³-1`.
+Look into the file named `/home/user/yaml/External Connections/DB Connections/MySQL.yaml`. It contains an entity of type `DbConnection` named `local`, with a field named `maxWait`. Its value is set to `9999999999999999999999999999999999`, and a `long` is expected by the model. Fix the field value to be an valid `long`. Max long value is `9.223372e+18 2⁶³-1`.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:32]: "a valid `long`" not "an valid `long`"

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 19]
```diff
@@ -103,33 +109,33 @@ TZVHO8mvbaG0weyJ9rQPOLXiZNwlz6bb65pcmaHFCN795trV1lpFDMS3wrUU77QR/w4VtfX128a9
...
 
-Look in the file named `/home/user/yaml/Policies/My policy/Sample policy.yaml`. An entity of type `FilterCircuit` named `Sample Policy` has a field named `category` that is pointing at an entity that does not exist in the configuration. Fix the YamlPk value in the field `category` to be something like `/System/Policy Categories/authentication`. In this case removing the field would also fix the issue as long as the default value of `/System/Policy Categories/miscellaneous` is what you want.
+Look into the file named `/home/user/yaml/Policies/My policy/Sample policy.yaml`. It contains an entity of type `FilterCircuit`, named `Sample Policy`, with a field named `category`, that is pointing to an entity that does not exists in the configuration. Fix the `YamlPk` value in the `category` field to something like `/System/Policy Categories/authentication`. In this case, removing the field would also fix the issue as long as the default value of `/System/Policy Categories/miscellaneous` is what you want.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:120]: "does not exist" not "does not exists"

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 20]
```diff
@@ -103,33 +109,33 @@ TZVHO8mvbaG0weyJ9rQPOLXiZNwlz6bb65pcmaHFCN795trV1lpFDMS3wrUU77QR/w4VtfX128a9
...
 
-If you are validating a configuration project that points to entities that are contained in another project that will be merged together later, you can use the `--allow-invalid-ref` parameter, to downgrade these issues from an ERROR to a WARNING in the trace file. The same message with a severity of WARNING will appear in this case.
+If you are validating a configuration project that points to entities that are contained in another project, that will be merged together later. You can use the `--allow-invalid-ref` parameter to downgrade these issues from an ERROR to a WARNING in the trace file. The same message with a severity of WARNING will show in this case.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:122]: The 1st 2 sentences should be 1 sentence as before or this makes no sense.

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 21]
```diff
@@ -103,33 +109,33 @@ TZVHO8mvbaG0weyJ9rQPOLXiZNwlz6bb65pcmaHFCN795trV1lpFDMS3wrUU77QR/w4VtfX128a9
...
+Look into the file named `/home/user/yaml/Policies/cert.yaml`. It contains an entity of type `GenerateSignatureFilter`, with a field named `signingCert`, which points to a certificate via a reference that does not exist in the configuration. The reference might be environmentalized using a YamlPK like this: `/Environment Configuration/Certificate Store/{{env "CERT"}}`, which has resulted in the "invalid field" in the error message. This issue occurs when you try to validate an environment where the system environment variable `CERT` does not exist. To avoid errors, use the `--allow-invalid-ref` option, or set a dummy value for the environment variable `CERT` in the validating environment.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:130]: "validate in an environment" not "validate an environment"
We are validating the config not the environment

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 22]
```diff
@@ -206,23 +212,23 @@ but should be changed to
...
 
 ## DbConnection - setPoolPreparedStatements
 
 **Severity**: WARNING
 
 `entity=/External Connections/DB Connections/MySQL/local, type=DbConnection, field='setPoolPreparedStatements', file=/home/user/yaml/External Connections/DB Connections/MySQL.yaml, message='Value will be set to 'false' but is neither falsy or truthy. Value: truex'`
 
-Look in the file named `/home/user/yaml/External Connections/DB Connections/MySQL.yaml`. It contains an entity of type `DbConnection` named `local` with a field named `setPoolPreparedStatements`. Its value is set to `truex` but an boolean is expected by the model. Fix the field value to be an valid boolean i.e. `true` or `false`. The value will default to `false`.
+Look into the file named `/home/user/yaml/External Connections/DB Connections/MySQL.yaml`. It contains an entity of type `DbConnection`, named `local`, with a field named `setPoolPreparedStatements`. Its value is set to `truex`, but a boolean is expected by the model. Fix the field value to be an valid boolean (`true` or `false`). The value defaults to `false`.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:231]: "to be a valid" not "to be an valid"

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 23]
```diff
@@ -103,33 +109,33 @@ TZVHO8mvbaG0weyJ9rQPOLXiZNwlz6bb65pcmaHFCN795trV1lpFDMS3wrUU77QR/w4VtfX128a9
...
+Look into the file named `/home/user/yaml/Policies/My policy/Sample policy.yaml`. It contains an entity of type `FilterCircuit`, named `Sample Policy`, with a field named `category`, which points to an entity that exists in the configuration but is of the wrong type. It is pointing to an entity of type `IPFilter`, but it needs an entity of type `PolicyCategory`. Fix the YamlPk value in the field `category` to something like `/System/Policy Categories/authentication`. In this case, removing the field would also fix the issue as long as the default value of `/System/Policy Categories/miscellaneous` is what you want.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:139]: I see YamlPk and YamlPK - let's use YamlPK everywhere ?

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md [hunk 24]
```diff
@@ -255,7 +260,7 @@ WARNING: Found entities of different types at same level with same PK for parent
 WARNING: Found entities of different types at same level with same PK for parent PK :'/Policies/Some Policies/Test' PK end with: (Reflector)Same name filter
 ```
 
-This is also generated at conversion time using the fed2yaml option. To fix after converting, edit the policy file in this case `/Policies/Some Policies/Test.yaml`, and rename one of the filters. Update all references to the filters so that the correct YamlPK is used:
+This is generated at conversion time using the `fed2yaml` option. To fix after converting, edit the policy file (`/Policies/Some Policies/Test.yaml`), and rename one of the filters.  Update all references to the filters so that the correct YamlPK is used:
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:263]: Why the ( and )

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 25]
```diff
@@ -38,42 +39,36 @@ The following is an example of environmentalization:
...
 fields:
    name: MySQL
    url: "jdbc:mysql://localhost:3306/DefaultDb"
-   password: HDidnjekj=
+   password: 'HDidnjekj='
    username: "scott"
 ```
 
-`{{ ... }}` placeholders are replaced when the configuration is loaded by the API Gateway or in ES Explorer.
+`{{...}}` placeholders are replaced when the configuration is loaded by the API Gateway or in ES Explorer.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:67]: Is it still  the case for ES Explorer ?

File: content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md (hunk 24 above)
Comment (andreamussap) [reply content/en/docs/apim_yamles/apim_yamles_references/yamles_known_validation_errors.md:263]: If you mean 'and rename one of the filters', this is what I could understand from the original sentence. If it's wrong, please tell me what's the correct text for this paragraph

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 26]
```diff
@@ -38,42 +39,36 @@ The following is an example of environmentalization:
...
+
+When you environmentalize references, the certificates also get environmentalized.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:69]: Needs rewording as references are not always certificates

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 27]
```diff
@@ -132,15 +127,54 @@ The following are elements of the environmentalization syntax.
...
+{{< alert title="Note">}}
+ Note that `{{ expression }}` (with spaces) works but is not recommended as those expression will be changed to `{{expression}}` at load time (due to parsing logic).
+ Thus when the YAML configuration is exported, it would create unexpected changes causing unwanted diffs in case the configuration is managed within Git.
+{{< /alert >}}
+
+#### Usage in YAML files
+
+All YAML files **must be parsable**.
+
+As `{` is interpreted it must be quoted -- only for the first char of a value.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:156]: I had to look at example to fully understand this sentence., maybe something like this...

When using `{{` in an environmentalized value always place the full value inside quotes when the value starts with `{{` to ensure proper interpretation. If the `{{` is embedded within the value, then quotes should not used (or are they optional) ???

What about 
"xxxxx{{db.host}}yyyy"
or
'xxx{{db.host}}yyy'

are they acceptable ?

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 28]
```diff
@@ -178,9 +212,11 @@ If none of the above conditions are true, an error is raised in the logs when th
 
 You can encode a string value in base64 using:
 
-```{{ base64 "changeme" }}```
+```{{base64 "changeme"}}```
 
-This is very useful for development or local environment, when your YAML configuration does not required to be encrypted. This will not work when your YAML configuration is encrypted as recommended for production environments.
+This is very useful for development or local environment, when your YAML configuration does not required to be encrypted. This will not work when your YAML configuration is encrypted, as recommended for production environments.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:217]: "is not required" instead of "does not required"

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 29]
```diff
@@ -269,12 +305,40 @@ value: 'a message with ''''single quotation marks'''' '
 
 ## Environmentalization in `values.yaml`
 
-To decouple where the value of an environment variable is coming from and where it is used, use the `{{ env "XYZ" }}` syntax in the `values.yaml` file:
+### Syntax
+
+As `{` is interpreted it must be quoted -- only for the first char of a value.
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:310]: Maybe reword to match above

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 30]
```diff
@@ -310,8 +372,8 @@ This can all be done without changing the YAML file for the entity that always p
 ---
 type: DBConnection
 fields:
-    # just showing environmentalization fields
-    host: {{ db.host }}
+    # this will will change depending on witch values.yaml is packaged
```
Comment (mcollins-axway) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:375]: "which" not "witch"

File: content/en/docs/apim_yamles/yamles_environmentalization.md (hunk 25 above)
Comment (bordigoni) [reply content/en/docs/apim_yamles/yamles_environmentalization.md:67]: Yes. It is the case. ESExplorer has not been changed, though the feature exists but applied yet.

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 31]
```diff
@@ -8,7 +8,7 @@
 
 The XML federated configuration provides different ways to environmentalize parts of the configuration. See [Environmentalize configuration](/docs/apigtw_devops/promotion_arch/#environmentalize-configuration) for details.
 
-The YAML configuration format replaces any environmentalization done via Policy Studio, which is enabled in Policy Studio through **Preferences > Environmentalization > Allow environmentalization**, with an improved method of environmentalization using the `{{ env "XYZ" }}` syntax. The conversion process will look after the replacement of this type of environmentalization. Selectors such as `${environment.XYZ}` and `${env.XYZ}` will continue to work as before.
+The YAML configuration format replaces any environmentalization done via Policy Studio, which is enabled in Policy Studio through **Preferences > Environmentalization > Allow environmentalization**, with an improved method of environmentalization using the `{{env "XYZ"}}` syntax. The conversion process will look after the replacement of this type of environmentalization. Selectors such as `${environment.XYZ}` and `${env.XYZ}` will continue to work as before.
```
Comment (andreamussap) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:11]: (which is enabled in Policy Studio through Preferences > Environmentalization > Allow environmentalization, with an improved method of environmentalization using the {{env "XYZ"}} syntax) --> can I remove this?

This makes the paragraph a bit confusing. You shouldn't document in here how to environmentalize in Policy studio, just link to the page where this is already documented, or maybe this isn't even needed because you've already linked to 'Environmentalize configuration' before ?

Comment (bordigoni) [reply content/en/docs/apim_yamles/yamles_environmentalization.md:11]: Policy Studio does not support YAML configurations yet.

The chapter is a reference to the PS feature we replace in with YAML-style environmentalization

Comment (andreamussap) [reply content/en/docs/apim_yamles/yamles_environmentalization.md:11]: @bordigoni how can I reword the paragraph? Can I remove that sentence?

File: content/en/docs/apim_yamles/yamles_environmentalization.md [hunk 32]
```diff
@@ -8,24 +8,26 @@
 
 The XML federated configuration provides different ways to environmentalize parts of the configuration. See [Environmentalize configuration](/docs/apigtw_devops/promotion_arch/#environmentalize-configuration) for details.
 
-The YAML configuration format replaces any environmentalization done via Policy Studio, which is enabled in Policy Studio through **Preferences > Environmentalization > Allow environmentalization**, with an improved method of environmentalization using the `{{env "XYZ"}}` syntax. The conversion process will look after the replacement of this type of environmentalization. Selectors such as `${environment.XYZ}` and `${env.XYZ}` will continue to work as before.
+The YAML configuration format replaces any environmentalization done via Policy Studio. The conversion process will look after the replacement of this type of environmentalization. Selectors such as `${environment.XYZ}` and `${env.XYZ}` will continue to work as before.
```
Comment (bordigoni) [inline content/en/docs/apim_yamles/yamles_environmentalization.md:11]: "_replaces any environmentalization done via Policy Studio_" is a little broad, YAML replaces the one you can activate with the "Allow environmentalization" menu, people know it also as the "earth symbol" you can find near fields you environmentalize in PS.
You could replace rephrase with :
"_replaces environmentalization done via Policy Studio (Earth symbol/icon)_"

**Review by andreamussap — APPROVED** [review]

**Comment by andreamussap** [comment]
@bordigoni @dominiquemongelli @mcollins-axway This was a huge team work. We've made lots of improvements to these contents. I'm merging the PR now, but let's keep the continuous improvements ball rolling! For any more updates or improvements, just create new PRs.
